cmd/snap-confine/mount-support.c

```diff
@@ -333,11 +334,34 @@ static void sc_bootstrap_mount_namespace(const struct sc_mount_config *config)
 						 config->rootfs_dir, dir);
 				sc_must_snprintf(dst, sizeof dst, "%s%s",
 						 scratch_dir, dir);
-				sc_do_mount(src, dst, NULL, MS_BIND, NULL);
-				sc_do_mount("none", dst, NULL, MS_SLAVE, NULL);
+				if (access(src, F_OK) == 0
+				    && access(dst, F_OK) == 0) {
+					sc_do_mount(src, dst, NULL, MS_BIND,
+						    NULL);
+					sc_do_mount("none", dst, NULL, MS_SLAVE,
+						    NULL);
+				}
 			}
 		}
 	}
+	if (config->uses_base_snap) {
+		// snapd dir
+		const char *src = "/usr/lib/snapd";
+		sc_must_snprintf(dst, sizeof dst, "%s%s", scratch_dir, src);
+		sc_do_mount(src, dst, NULL, MS_REC | MS_BIND, NULL);
+		sc_do_mount("none", dst, NULL, MS_REC | MS_SLAVE, NULL);
+
+		// FIXME: snapctl tool - our apparmor policy wants it in
```

Comment:
How about this: let's always use `/var/lib/snapd/tools` as a place where snapd keeps it tooling. Snap confine could then mount tmpfs there and create a symlink or bind mount farm from each tool (wherever it is, even from the file that is invisible in the base snap) to an empty file here.

As long as we can require /var/lib/snapd to exist in base snaps we could then run our tools reliably.